src/app/services/utils.service.ts

```diff
@@ -178,23 +170,56 @@ export class UtilsService {
         }).format(date);
 
       default:
-        return new Intl.DateTimeFormat('en-GB', {
-          hour12: true,
-          hour: 'numeric',
-          minute: 'numeric',
-          month: 'short',
+      return this.dateFormatter(date, true);
+    }
+  }
+
+  dateFormatter(date: Date, isDuedate?: boolean) {
+    const today = new Date();
+    let formattedDate = new Intl.DateTimeFormat('en-GB', {
+      month: 'short',
+      day: 'numeric',
+      year: 'numeric'
+    }).format(date);
+
+    if (date.getFullYear() === today.getFullYear() && date.getMonth() === today.getMonth()) {
+      if (date.getDate() === today.getDate() - 1) {
+        formattedDate = 'Yesterday';
+      }
+      if (date.getDate() === today.getDate()) {
+        formattedDate = 'Today';
+      }
+      if (date.getDate() === today.getDate() + 1) {
+        formattedDate = 'Tomorrow';
+      }
+      if (isDuedate && date.getDate() > today.getDate() + 1) {
+        formattedDate = new Intl.DateTimeFormat('en-GB', {
+          month: 'numeric',
           day: 'numeric',
           year: 'numeric'
         }).format(date);
+      }
     }
+
+    if (isDuedate) {
+      formattedDate += ' ' + new Intl.DateTimeFormat('en-GB', {
+        hour12: true,
+        hour: 'numeric',
+        minute: 'numeric'
+      }).format(date);
+    }
+    return formattedDate;
   }
 
-  timeComparer(timeString: string, comparedString?: string) {
+  timeComparer(timeString: string, comparedString?: string, dueToday?: boolean) {
```

Comment:
this is `utils.service` so everything should be generic, `dueToday` is not a generic term here.
So the logic of this function is to compare two times. Instead of `dueToday`, we can use something like `compareDate?: boolean` to let this function compare the date difference instead of time difference. That will give us the information we need.